smart-contracts/test/BridgeProxy/ArbitrumOrbitNativeDepositBridgeProxy.ethereum.ts

```diff
@@ -0,0 +1,324 @@
+  data: string
+) => {
+  const destProvider = await getProvider(destinationChainId)
+
+  // You will need to implement these contract ABIs and addresses yourself
+  const { interface: NodeInterface } = await ethers.getContractAt(
+    'INodeInterface',
+    ZeroAddress
+  )
+  const nodeInterface = new ethers.Contract(
+    '0x00000000000000000000000000000000000000C8',
+    NodeInterface,
+    destProvider
+  )
+
+  // 1. Estimate gasLimit for the retryable ticket
+  const [signer] = await ethers.getSigners()
+  const gasLimit = await nodeInterface.estimateRetryableTicket.estimateGas(
+    await signer.getAddress(), // from,
+    amount + ethers.parseEther('1'), // senderDeposit, // can be a dummy value (e.g., l2CallValue + 1 ether)
+    await bridge.L1_BRIDGE_PROXY(), // to
+    amount, // l2CallValue
+    await bridge.getAddress(), // excessFeeRefundAddress
+    await bridge.getAddress(), // callValueRefundAddress (receives msg.value on l2)
+    data // data
+  )
+
+  const { maxFeePerGas, maxSubmissionCost } =
+    await estimateRetryableFee(destinationChainId)
+
+  // 4. Compute deposit (sum of all costs, plus l2CallValue)
+  const deposit = gasLimit * maxFeePerGas! + maxSubmissionCost + amount
+
+  // 5. Return all values as an object (in BN/BigInt/string as needed)
+  return {
+    deposit,
+    gasLimit,
+    maxFeePerGas,
+    maxSubmissionCost,
+  }
+}
+
+describe('ArbitrumOrbitNativeBridgeProxy (deposit)', function () {
+  let bridge: any
+  let recipient: Signer
+
+  before(async () => {
+    ;[, recipient] = await ethers.getSigners()
+
+    // deploy using ignition
+    const parameters = {
+      ArbitrumOrbitNativeDepositBridgeProxy: {
+        inbox,
+        l1BridgeProxy,
+        relayPool,
+        relayPoolChainId: destinationChainId,
+        routerGateway,
+      },
+    }
+
+    const result = await ignition.deploy(
+      ArbitrumOrbitNativeDepositBridgeProxyModule,
+      {
+        parameters,
+      }
+    )
+    bridge = result.bridge
+  })
+
+  describe.only('sending ERC20', () => {
+    let balanceBefore: bigint
+    const amount = parseUnits('0.1', 18)
+    let receipt: TransactionReceipt | null
+
+    before(async () => {
+      // fund assets to the bridge contract
+      await stealERC20(
+        UDT_ETHEREUM,
+        UDT_WHALE,
+        await bridge.getAddress(),
+        amount
+      )
+
+      // get initial balance
+      balanceBefore = await getBalance(
+        await bridge.getAddress(),
+        UDT_ETHEREUM,
+        ethers.provider
+      )
+
+      // make sure stealing pays
+      expect(balanceBefore).to.equal(amount)
+
+      // TODO: clarify how to calculate / simualte this calldata
+      // and compute gaslimit
+      const gasLimit = 6000000n
+      const calldataLength = 1652
+      const fakeCalldata = Array(calldataLength).fill(0).join('')
+
+      const { maxFeePerGas, maxSubmissionCost } = await estimateRetryableFee(
+        destinationChainId,
+        fakeCalldata
+      )
```

Comment:
couldnt really figure out the correct way to estimate gas for ERC20 so this is ballpark -  the arb sdk works only with ethers5